frontend/staff-dashboard/src/components/flexiblepricing/statusmodal.tsx

```diff
@@ -42,7 +42,8 @@ export const FlexiblePricingStatusModal: React.FC<IFlexiblePriceStatusModalProps
     const mutationResult = useUpdate<IFlexiblePriceRequest>();
     const { mutate } = mutationResult;
 
-    const [ justification, setJustification ] = useState(modaldata.justification);
+    let [ justification, setJustification ] = useState(modaldata.justification);
+    justification = justification || status !== "approved" ? justification : "Documents in order"
```

Comment:
If this is using React state, then you need to be using the setJustification function so reactivity works right.

Reply:
Makes sense. Will do it this way.